src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -227,11 +227,34 @@ double FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::Evaluate(
 
 template<typename OutputLayerType, typename InitializationRuleType,
          typename... CustomLayers>
-void FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::Gradient(
-    const arma::mat& parameters,
-    const size_t begin,
-    arma::mat& gradient,
-    const size_t batchSize)
+double FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::Evaluate(
+    const arma::mat& parameters, const size_t begin, const size_t batchSize)
+{
+  return Evaluate(parameters, begin, batchSize, true);
+}
+
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+template<typename GradType>
+double FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::
+EvaluateWithGradient(const arma::mat& parameters, GradType& gradient)
+{
+  double res = 0;
+  for (size_t i = 0; i < predictors.n_cols; ++i)
+    res += EvaluateWithGradient(parameters, i, gradient, 1, false);
```

Comment:
Almost certainly a larger batch size will help this be faster.  I'd suggest a default of 64, 128, or 256 (those seem to be common values).

Reply:
This default seems to originate from the time when only stochastic models were there. I'll choose the lowest default of 64, as 256 might be too much.

Reply:
@rcurtin On another note, I'm not too sure if this should be changed or not. Many old tests would have been implemented keeping stochastic training in mind, and we anyways provide the option for `batchSize` now. So I don't see a point meddling with the old tests.

Reply:
Shouldn't the result be exactly the same?  Or do you mean that there are models for which simply the computation of the gradient or objective function changes the parameters of the model?

Reply:
I just want this to follow from the `Evaluate()` implementation which does a stochastic evaluation only. Choosing another default would be a break from that design.